### PR TITLE
fix #12145: move mapsforge svg to own dir, delete leftovers, patch mapsforge

### LINKED ISSUE #12145: Thousands of SVG cache files
### Describe your problem!

I've got 63,702 PNG files named `svg-\d+.png` in `/data/data/cgeo.geocaching/files`. They contain icons. Dates range from the present back to beginning of the year.

### How to reproduce?

…

### Actual result after these steps?

_No response_

### Expected result after these steps?

_No response_

### Reproducible

Unclear

### c:geo Version

Doesn't matter, various versions

### System information

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by eddiemuc**
fix #12145: move mapsforge svg to own dir, delete leftovers, patch mapsforge

### CODE REVIEW

**Comment by eddiemuc** [comment]
Update: mapsforge patch (in cgeo/mapsforge) is now included

**Comment by eddiemuc** [comment]
This PR got lucky too - passed CI on first try :-)

**Comment by eddiemuc** [comment]
Update: corresponding PR on mapsforge was already merged to their master. Very quick reaction over there! 

This means that on next mapsforge release we can safely switch to it (and abandoning current special build).

**Comment by Lineflyer** [comment]
@eddiemuc 
Release is done, which means we can merge this to `release`, can't we?

**Comment by eddiemuc** [comment]
Yes we can :-)

File: main/src/cgeo/geocaching/storage/LocalStorage.java [hunk 1]
```diff
@@ -46,6 +46,7 @@
     private static final String GEOCACHE_DATA_DIR_NAME = "GeocacheData";
     private static final String OFFLINE_LOG_IMAGES_DIR_NAME = "OfflineLogImages";
     private static final String MAP_THEME_INTERNAL_DIR_NAME = "MapThemeData";
+    private static final String MAPSFORGE_SVG_CACHE_DIR_NAME = "mapsforge-svg-cache";
     private static final long LOW_DISKSPACE_THRESHOLD = 1024 * 1024 * 100; // 100 MB in bytes
```
Comment (Lineflyer) [inline main/src/cgeo/geocaching/storage/LocalStorage.java:50]: Sorry for asking to late:
Is this still inside some user-inaccessible area or does this target to our base dir or additional geocache data directory?
IMHO those are cache files and should stay internal (isn't there a specific cache dir assigned to each app by the system).

Comment (eddiemuc) [reply main/src/cgeo/geocaching/storage/LocalStorage.java:50]: Still inside the internal dir but not in the special android cache dir. Same like all the other dirs in the list.

The idea to move stuff to Android cache dir is worth exploring (there's more candidates than just the mapsforge SVG), but maybe worth its own issue

Comment (fm-sys) [reply main/src/cgeo/geocaching/storage/LocalStorage.java:50]: > The idea to move stuff to Android cache dir is worth exploring (there's more candidates than just the mapsforge SVG), but maybe worth its own issue


Yes, that would probably be a good idea!

Even though I have reinstalled cgeo recently, there's already 1.5GB content in the cgeo internal directory again.

Using the caching directory would have the benefit of having an easy way to delete cached files without touching important data like preferences or the DB.

Screenshot_20211122_073508_com android settings_copy_432x936

Comment (eddiemuc) [reply main/src/cgeo/geocaching/storage/LocalStorage.java:50]: I believe the vast majority of the internal space is consumed by downloaded cache data in dir `GeocacheData` (especially images from listings). The cache directory seems to contains just data from Google Maps, leakcanary and some debug stuff.

Comment (Lineflyer) [reply main/src/cgeo/geocaching/storage/LocalStorage.java:50]: @fm-sys 
AFAIK these figures include the "additional geocache data" located in user readable store but not only the internal data (like theme cache, database, etc.)

Comment (eddiemuc) [reply main/src/cgeo/geocaching/storage/LocalStorage.java:50]: @Lineflyer I checked on my device. I have around 5000 caches stored offline, and the total space consumed shown is about 1GB. This is around 200KB per cache. c:geo downloads and stores all images of a listing in the GeocacheData dir (together with some meta-info e.g. etag and last-modified), and this data alone should make up the larger part of the 200KB in average. If someone turns on saving of log images, the data size will surely become much more

Why not put this data in the cache dir?

Comment (fm-sys) [reply main/src/cgeo/geocaching/storage/LocalStorage.java:50]: > @fm-sys
AFAIK these figures include the "additional geocache data" located in user readable store but not only the internal data (like theme cache, database, etc.)

The external/public directory is more than 3GB on my device cause of quite some offline maps. So I believe this folder isn't included in these figures on my device...

> and this data alone should make up the larger part of the 200KB in average. If someone turns on saving of log images, the data size will surely become much more
>
>Why not put this data in the cache dir?

At least if the setting is enabled, pictures shouldn't be deleted automatically by the system. But in all other cases these files should IMHO go into the app-cache directory...

Comment (Lineflyer) [reply main/src/cgeo/geocaching/storage/LocalStorage.java:50]: Lets take the topic "Usage of system cache dir" into an own issue. Will open it.

**Pull request merged**